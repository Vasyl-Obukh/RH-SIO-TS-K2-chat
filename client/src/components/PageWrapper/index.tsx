import * as React from 'react';
import Header from '../Header';
import Footer from '../Footer';

interface Props {
  component: React.FC
}

function PageWrapper({ component: Component }: Props) {
  return (
    <>
      <Header />
      <Component />
      <Footer />
    </>
  );
}

export default PageWrapper;