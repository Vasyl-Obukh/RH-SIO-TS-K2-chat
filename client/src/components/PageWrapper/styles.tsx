import styled from 'styled-components';

export const ComponentContainer = styled.div`
  padding: 25px 50px;
`;