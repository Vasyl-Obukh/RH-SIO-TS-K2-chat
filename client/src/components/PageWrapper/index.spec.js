import React from 'react';
import { shallow } from 'enzyme';
import PageWrapper from './index';

describe('<PageWrapper /> tests', () => {
  const testString = 'test';

  test('Should render correctly', () => {
    const component = shallow(<PageWrapper component={testString} />);
    expect(component).toMatchSnapshot();
  });

  test('', () => {
    const component = shallow(<PageWrapper component={testString} />);
    const stringNodes = component.findWhere(n => n.getElement().type == 'test');
    expect(stringNodes.length).toBe(1);
  });
});