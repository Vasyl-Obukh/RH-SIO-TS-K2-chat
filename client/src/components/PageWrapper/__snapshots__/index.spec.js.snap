// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageWrapper /> tests Should render correctly 1`] = `
<Fragment>
  <Header />
  <test />
  <Footer />
</Fragment>
`;
